src/include/platform/DiagnosticDataProvider.h

```diff
@@ -103,14 +103,82 @@ class DiagnosticDataProvider
     /**
      * General Diagnostics methods.
      */
+
+    /**
+     * @brief Obtain the number of times the node has rebooted.
+     *
+     * @param[out] rebootCount Reference to location where the reboot count integer will be copied.
+     * @returns CHIP_NO_ERROR on success or another CHIP_ERROR from the underlying implementation
```

Comment:
This `@returns` comment is not communicating anything the return type is not already telling you.  Please just take it, and similar comments below, out, because they are just taking up space without saying anything.

Reply:
I see the redundancy. My reference was `DeviceInstanceInfoProvider.h`. Is there a standard regarding API documentation? I saw different formats across multiple header files.

Reply:
I don't know that there's anything formal written down, and we have a lot of existing bad documentation.  :(

The basic guideline I would use is that the documentation should tell you the things you can't tell by just looking at the method signature.